mono/metadata/mono-perfcounters.c

```diff
@@ -743,25 +743,20 @@ instance_search (SharedHeader *header, void *data)
 static SharedInstance*
 find_custom_instance (SharedCategory* cat, char *name)
 {
-	InstanceSearch search;
+	InstanceSearch search = { 0 };
```

Comment:
IMO, since we're using C99 these days, an even better style would be:

```c
InstanceSearch search = {
    .cat_offset = (char *) cat - (char *) shared_area,
    .cat = cat,
    .name = name,
};
```

When using designated initializers like this, any field that isn't mentioned will be default-initialized.

Reply:
Not yet implemented by Microsoft.

Reply:
Does this zero the other fields? It does with my style, with the union caveat. Otherwise, yes, it seems like an obvious language feature they all should have added several decades ago..

Reply:
> Not yet implemented by Microsoft.

It is. We use this in the code base already.

> Does this zero the other fields?

Yes.

Reply:
I couldn't this to compile as C++, the designated initializer. {0} works.

Reply:
Designated initializers are not standard C++ either. It may or may not work as a compiler extension. Why do we care if it's valid C++ though?

Reply:
In dream of some day compiling as C++.